torchtune/data/_messages.py

```diff
@@ -463,3 +463,45 @@ def __call__(self, sample: Mapping[str, Any]) -> Mapping[str, Any]:
             updated_messages.append(Message.from_dict(message))
 
         return {"messages": updated_messages}
+
+
+def validate_messages(
```

Comment:
If we're moving this here, should also delete from data/_utils.py?

Reply:
Also for multimodal, do we need to do any extra validation? E.g. number of images == number of messages with type == 'image'? (Not saying we should do it here btw)

Reply:
good point... I am not sure yet where this should happen. maybe note this as a followup once the whole end-to-end is established